### ISSUE TITLE
publish a public prototype

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/natural-selection#116]
@ariel-phet has mentioned that it would be nice to publish a prototype version of this sim, before it is completed.

Questions:

* When (ideally) would you like the prototype published?
* What things do we need to complete?
* What is the process for publishing a prototype?

Assigning to the design team for feedback.

**Comment by ariel-phet** [comment]
@pixelzoom I think "ideally" we would publish a preview version by Aug 15

- Ideally the sim would be basically feature complete (visual cleanliness not being the priority)
- There should be no "obvious" bugs, the sim basically runs and meets the learning goals
- For example if the population graph was working, but wasn't as polished as we like that would not be an issue
- If there are some undesirable but not "showstopper" performance issues (particularly on older platforms) that is fine
- The code does not need to be reviewed or even have had an internal review done

The process for publishing a preview is to have a few of us look at it (like Amanda, myself, Oliver) and make sure it is in reasonable shape, and then I believe a "dev" version is published, and handed to @mattpen - he manually puts it on the web at the placeholder address where the "latest" html version will eventually live. 

For example the prototype of AC CCK lives here:

http://phet.colorado.edu/sims/html/circuit-construction-kit-ac/latest/circuit-construction-kit-ac_en.html

but no sim page yet exists, just this direct link.

**Comment by kathy-phet** [comment]
Yes, this would be nice. Ariel summed it up nicesly.

**Comment by amanda-phet** [comment]
That all sounds right to me as well.

**Comment by mattpen** [comment]
> What is the process for publishing a preview version?

Currently the process is "Ask Matt and he'll work some magic".  I was under the impression that this was a one-off request for the 4 sims published early during the Covid-19 shutdown, so it isn't documented well or easy for others to replicate. If this is going to be a regular feature we want to include on the website, we should probably formalize/standardize a process that allows sim developers to publish these without coordination with the website team.

**Comment by pixelzoom** [comment]
> What things do we need to complete?

I appreciate the general description, which could apply to any sim. But I need specifics for this sim.  The sim is essentially feature complete now, so run master or [1.0.0-dev.66](https://phet-dev.colorado.edu/html/natural-selection/1.0.0-dev.66/phet/natural-selection_en_phet.html), look at the open GitHub issues, and please put together a checklist of what needs to be completed before publishing a preview.

**Comment by ariel-phet** [comment]
This won't need @kathy-phet approval, removing her assignment. @amanda-phet can be the main arbiter of what is an acceptable state, I will also try to take a look.

**Comment by ariel-phet** [comment]
A quick scan of the github issues, I think only #98 and #86 are essential to get to an acceptable place before publishing a prototype version, most of the other issues seem performance or iO related and not critical.

**Comment by pixelzoom** [comment]
I also want to very #46 and #112, memory consumption issues.

**Comment by amanda-phet** [comment]
@ariel-phet do you think we should do interviews before publishing the public preview version? It will significantly delay the preview version, but still seems worth considering.

**Comment by pixelzoom** [comment]
@amanda-phet and I discussed...  Other than the interviews question that @amanda-phet asked in https://github.com/phetsims/natural-selection/issues/116#issuecomment-665189886, we feel like the remaining GitHub issues do not need to be addressed for a public preview.   

@ariel-phet and @kathy-phet how do you feel about the state of the sim? Are we ready for a public preview?

**Comment by ariel-phet** [comment]
@amanda-phet @pixelzoom definitely ready for a public preview, no need to ask @kathy-phet and interviews are not necessary for this preview. Especially since this sim is based off a previous sim, I am pretty confident everything is an upgrade. 

@pixelzoom - I need to speak with QA about https://github.com/phetsims/perennial/issues/190 but I think we should move ahead with following the RC process. I will speak with @KatieWoe about that today

**Comment by pixelzoom** [comment]
> I need to speak with QA about phetsims/perennial#190 but I think we should move ahead with following the RC process.  I will speak with @KatieWoe about that today

@ariel-phet please clarify. This is not an RC process, and QA is not involved in preview versions. My understanding is that this is a dev version that @mattpen will manually put on the web at the placeholder address where the "latest" html version will eventually live.

**Comment by pixelzoom** [comment]
Ah... I see in https://github.com/phetsims/perennial/issues/190#issuecomment-659617161 that there will be some type of testing by QA for preview version (or prototype versions --- is that the same thing?).

**Comment by pixelzoom** [comment]
I've changed "preview" to "prototype" throughout the comments in this issue, since that's the name that we decided on for this artifact.

**Comment by pixelzoom** [comment]
In a Slack discussion, @ariel-phet clarified that a prototype will be published using the RC+production process, with a level of testing that's more typical of a dev version.  I'll await the process documentation in [chipper-2.0.md](https://github.com/phetsims/phet-info/blob/master/deployment-info/chipper-2.0.md), which https://github.com/phetsims/perennial/issues/190#issuecomment-659617161 indicates @ariel-phet will be writing up.

**Comment by pixelzoom** [comment]
[1.0.0-rc.1](https://phet-dev.colorado.edu/html/natural-selection/1.0.0-rc.1/phet/natural-selection_en_phet.html) was published using the process described in https://github.com/phetsims/phet-info/blob/master/deployment-info/chipper-2.0.md#prototype-deployments.  The QA issue is https://github.com/phetsims/QA/issues/522.

**Comment by pixelzoom** [comment]
As noted in https://github.com/phetsims/QA/issues/522#issuecomment-669640638:

> In retrospect we should have done a dev test first. There have been a lot of changed required for this RC. And it has become increasingly difficult (and time-consuming) to patch changes into the 1.0 branch, to the point where forward progress on master has ground to a halt. After consulting with @ariel-phet, I'm going to give up on trying to patch the 1.0 branch, and only make changes in master. The next RC will either be in a new 1.1 branch, or I will merge all changes from master into 1.0. And before that RC, we'll do a quick test on a dev version.

**Comment by pixelzoom** [comment]
In https://github.com/phetsims/natural-selection/issues/116#issuecomment-655713562 @ariel-phet said
> @pixelzoom I think "ideally" we would publish a prototype by Aug 15

Acknowledgement that we have missed that milestone.  @amanda-phet @ariel-phet @kathy-phet Do we want to set a new milestone?

**Comment by pixelzoom** [comment]
8/18/2020 design meeting: The new milestone for the prototype is "sometime next week if possible".

**Comment by pixelzoom** [comment]
8/20/2020 design meeting: 

The new plan for the prototype (brand=phet) is:

* dev test master
* create a new 1.1 release branch, publish RC 1.1.0-rc.1
* RC will have "light testing" - no QA, light testing by @amanda-phet @ariel-phet @oliver-phet
* prototype will be made visible on the PhET website

Then the sim will go through a full QA process for both brands, and will be republished.

**Comment by pixelzoom** [comment]
We did not meet the milestone of "sometime the week of 8/24, if possible".

Assigning to @amanda-phet @kathy-phet @ariel-phet to revise the milestone.

**Comment by pixelzoom** [comment]
8/27/2020 design meeting:

All blocking issues are addressed, ready to publish a dev version for prototype. 

The plan:

* Publish dev version today Thursday 8/27. Assign to @phet-steele @amanda-phet @ariel-phet @oliver-phet to test. @ariel-phet will do string tests. Testing of dev version to be completed by end of business Friday 8/28.  
* Monday 8/31 I will publish RC version 1.1.0-rc.1 using dev versions shas, have the same people spot test by Tuesday 9/1 noon.
* Publish 1.1.0.   
* Adjust schedule if any issues are found.
* @oliver-phet will prep an email announcement. @ariel-phet will announce on social media.

**Comment by pixelzoom** [comment]
[1.1.0-dev.18](https://phet-dev.colorado.edu/html/natural-selection/1.1.0-dev.18/phet/natural-selection_all_phet.html) has been published. The QA issue is https://github.com/phetsims/QA/issues/539.

**Comment by pixelzoom** [comment]
[1.1.0-rc.1](https://phet-dev.colorado.edu/html/natural-selection/1.1.0-rc.1/phet/natural-selection_en_phet.html) has been published. The QA issues is https://github.com/phetsims/QA/issues/540.

**Comment by pixelzoom** [comment]
[1.1.0-rc.2(https://phet-dev.colorado.edu/html/natural-selection/1.1.0-rc.2/phet/natural-selection_en_phet.html) has been published. The QA issue is https://github.com/phetsims/QA/issues/540.

**Comment by ariel-phet** [comment]
This is going to be considered a production version when put on the website, so we can close this issue (even though it got a more "prototype" QA process).

**Comment by pixelzoom** [comment]
[1.1.0](https://phet-dev.colorado.edu/html/natural-selection/1.1.0/phet/natural-selection_en_phet.html) was published today, 9/1/2020.
